tidoop-mr-lib-api/src/mysql_driver.js

```diff
@@ -0,0 +1,123 @@
+        var query = connection.query(
+            'INSERT INTO tidoop_job (jobId, jobType, startTime, mapProgress, reduceProgress) ' +
+            'VALUES (?, ?, NOW(), ?, ?)',
+            [jobId, jobType, 0, 0],
+            function (error, result) {
+                if (error) {
+                    callback(error)
+                } else {
+                    console.log('Successful insert: \'INSERT INTO tidoop_job ' +
+                        '(jobId, jobType, startTime, mapProgress, reduceProgress) VALUES' +
+                        '(' + jobId + ', ' + jobType + ', NOW(), 0, 0)\'');
+                    callback(null, result);
+                } // if else
+            }
+        );
+    }, // addJob
+
+    updateJobStatus: function (jobId, mapProgress, reduceProgress, callback) {
+        if (mapProgress == 100 && reduceProgress == 100) {
+            var query = connection.query(
+                'UPDATE tidoop_job SET endTime=NOW(), mapProgress=?, reduceProgress=? WHERE jobId=\'' + jobId + '\'',
+                [mapProgress, reduceProgress],
+                function (error, result) {
+                    if (error) {
+                        callback(error);
+                    } else {
+                        console.log('Successful update: \'UPDATE tidoop_job ' +
+                            'SET endTime=NOW(), mapProgress=' + mapProgress + ', reduceProgress=' +
+                            reduceProgress + ' WHERE jobId=\'' + jobId + '\'\'');
+                        callback(null, result);
+                    } // if else
+                }
+            );
+        } else {
+            var query = connection.query(
+                'UPDATE tidoop_job SET mapProgress=?, reduceProgress=? WHERE jobId=\'' + jobId + '\'',
+                [mapProgress, reduceProgress],
+                function (error, result) {
+                    if (error) {
+                        callback(error);
+                    } else {
+                        console.log('Successful update: \'UPDATE tidoop_job ' +
+                            'SET mapProgress=' + mapProgress + ', reduceProgress=' + reduceProgress +
+                            ' WHERE jobId=\'' + jobId + '\'\'');
+                        callback(null, result);
+                    } // if else
+                }
+            );
+        } // if else
+    }, // updateJobStatus
+
+    getJob: function (jobId, callback) {
+        var query = connection.query(
+            'SELECT * from tidoop_job WHERE jobId=\'' + jobId + '\'',
+            function (error, result) {
+                if (error) {
+                    callback(error);
+                } else {
+                    console.log('Successful select: \'SELECT * from tidoop_job WHERE jobId=\'' + jobId + '\'\'');
+                    callback(null, result);
+                } // if else
+            }
+        );
+    }, // getJobStatus
+
+    close: function(connection) {
+        connection.end();
```

Comment:
Since the `end()` function seems to be asynchronous in the module you are using (see https://github.com/felixge/node-mysql/#terminating-connections), I would make your close asynchronous too to let the calling part deal with any error which may arise, this is:
```
close: function(connection, callback) {
  connection.end(callback);
});
```